frog/imports/client/Wiki/index.js

```diff
@@ -182,12 +194,76 @@ class WikiComp extends React.Component<WikiCompPropsT, WikiCompStateT> {
       );
     }
 
+    // Show locked modal if the wiki is locked
+    if (
+      this.wikiDoc.data.settings?.locked &&
+      !this.wikiDoc.data.owners.find(x => x === Meteor.userId())
+    ) {
+      this.setState({ currentPageObj: null });
+      this.props.showModal(<LockedModal />);
+      return;
+    } else {
+      this.initialLoad = true;
+      this.props.hideModal();
+    }
+    // Ask for password if wiki access is password restricted
+    if (
+      this.wikiDoc.data.settings?.restrict === 'view' &&
+      !this.wikiDoc.data.users?.find(x => x === Meteor.userId()) &&
+      !this.wikiDoc.data.owners.find(x => x === Meteor.userId())
+    ) {
+      this.setState({ currentPageObj: null });
+      const passwordPromise = new Promise(resolve => {
+        this.props.showModal(
+          <PasswordModal
+            callback={resolve}
+            hideModal={this.props.hideModal}
+            actualPassword={this.wikiDoc.data.settings?.password}
+          />
+        );
+      });
+      const result = await passwordPromise;
+      if (!result) {
+        this.props.showModal(
```

Comment:
We should move this modal to its own file to reduce code complexity and remove the dependency of `index.js` on Modal

Reply:
Why should we create a new file for 10 lines of code?